tests/framework/extensions/clusters/aks/resources/nodepools.go

```diff
@@ -0,0 +1,73 @@
+package resources
+
+import (
+	"time"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters/aks"
+	"github.com/sirupsen/logrus"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	active = "active"
+)
+
+// updateNodePoolQuantity is a helper method that will update the node pool with the desired quantity.
+func updateNodePoolQuantity(client *rancher.Client, cluster *management.Cluster, nodePool *aks.NodePool) (*management.Cluster, error) {
+	clusterResp, err := client.Management.Cluster.ByID(cluster.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	var aksConfig = clusterResp.AKSConfig
+	*aksConfig.NodePools[0].Count += *nodePool.NodeCount
```

Comment:
Is it possible to have a match function like we have for the rancher clusters? I think that'd make these functions more flexible

Reply:
The difference with the rancher clusters vs the hosted clusters is there is no concept of a node role. For AKS, for example, Eech cluster is going to be recognized as a worker. So that's a key difference as to why it's not as flexible